src/main/java/com/sourcery/gymapp/backend/workout/mapper/RoutineExerciseMapper.java

```diff
@@ -0,0 +1,58 @@
+package com.sourcery.gymapp.backend.workout.mapper;
+
+import com.sourcery.gymapp.backend.workout.dto.CreateRoutineExerciseDto;
+import com.sourcery.gymapp.backend.workout.dto.ExerciseSimpleDto;
+import com.sourcery.gymapp.backend.workout.dto.ResponseRoutineExerciseDto;
+import com.sourcery.gymapp.backend.workout.model.Exercise;
+import com.sourcery.gymapp.backend.workout.model.Routine;
+import com.sourcery.gymapp.backend.workout.model.RoutineExercise;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RoutineExerciseMapper {
+
+    public RoutineExercise toEntity(
+            CreateRoutineExerciseDto createRoutineExerciseDto,
+            Routine routine,
+            Exercise exercise) {
+
+        RoutineExercise routineExercise = new RoutineExercise();
+        routineExercise.setId(createRoutineExerciseDto.exerciseId());
+        routineExercise.setExercise(exercise);
+        routineExercise.setRoutine(routine);
+        routineExercise.setOrderNumber(createRoutineExerciseDto.orderNumber());
+        routineExercise.setDefaultSets(createRoutineExerciseDto.defaultSets());
+        routineExercise.setDefaultReps(createRoutineExerciseDto.defaultReps());
+        routineExercise.setDefaultWeight(createRoutineExerciseDto.defaultWeight());
+        routineExercise.setDefaultRestTime(createRoutineExerciseDto.defaultRestTime());
+        return routineExercise;
+    }
+
+    public CreateRoutineExerciseDto toCreateRoutineExerciseDto(RoutineExercise routineExercise) {
+        return new CreateRoutineExerciseDto(
+                routineExercise.getId(),
+                routineExercise.getOrderNumber(),
+                routineExercise.getDefaultSets(),
+                routineExercise.getDefaultReps(),
+                routineExercise.getDefaultWeight(),
+                routineExercise.getDefaultRestTime(),
+                routineExercise.getNotes()
+        );
+    }
+
+    public ResponseRoutineExerciseDto toResponseRoutineExerciseDto(RoutineExercise routineExercise) {
+        return new ResponseRoutineExerciseDto(
+                routineExercise.getId(),
+                new ExerciseSimpleDto(
+                        routineExercise.getExercise().getId(),
```

Comment:
It's better to create `new ExerciseSimpleDto` first and then pass it `return new ResponseRoutineExerciseDto()` for  better readability

Reply:
Hmmm not sure, I think its quite readable but I can change that if you say so